quarkus/config-api/src/main/java/org/keycloak/config/StorageOptions.java

```diff
@@ -17,13 +17,145 @@
+            .build();
+
+    public static final Option<String> STORAGE_ROLE = new OptionBuilder<>("storage-role", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_USER = new OptionBuilder<>("storage-user", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_DEPLOYMENT_STATE = new OptionBuilder<>("storage-deployment-state", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_AUTH_SESSION = new OptionBuilder<>("storage-auth-session", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_USER_SESSION = new OptionBuilder<>("storage-user-session", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_LOGIN_FAILURE = new OptionBuilder<>("storage-login-failure", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_AUTHORIZATION_PERSISTER = new OptionBuilder<>("storage-authorization-persister", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_USER_SESSION_PERSISTER = new OptionBuilder<>("storage-user-session-persister", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_ACTION_TOKEN = new OptionBuilder<>("storage-action-token", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_DBLOCK = new OptionBuilder<>("storage-dblock", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_CACHE_REALM_ENABLED = new OptionBuilder<>("cache-realm-enabled", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> STORAGE_CACHE_USER_ENABLED = new OptionBuilder<>("cache-user-enabled", String.class)
+            .category(OptionCategory.STORAGE)
+            .hidden()
+            .buildTime(true)
+            .build();
```

Comment:
Yes, we need to have them to be able to map accordingly.

In fact, that is something I'm evaluating because the current status of the new store is not yet disabling the legacy-related components completely, as we discussed. That was a bit surprising.

That said, I think the best here is to make legacy components environment-dependent and check whether or not the map storage is supported.

Wdyt?

Reply:
As discussed, I'll be sending the changes to disable legacy-related providers in another round of changes.